### PR TITLE
Core 1494 - Fix data search for logged out users.

### PR DESCRIPTION
**Pull request by sri2k1us**
- Fetch bootstrap info if it not available cache
- Restrict logged out data search results to `Community Data` folder.

### CODE REVIEW

File: src/components/search/dataSearchQueryBuilder.js [hunk 1]
```diff
@@ -9,12 +11,13 @@ export const getDataSimpleSearchQuery = (
...
         query: {
-            all: userHomeDir
+            all: !isDetailed
                 ? searchClauses.concat({
                       type: "path",
-                      args: { prefix: userHomeDir },
+                      args: { prefix: pathPrefix },
                   })
                 : searchClauses,
         },
```
Comment (sri2k1us) [inline src/components/search/dataSearchQueryBuilder.js:23]: For detailed search page, we don't want restrict results to any particular path.

Comment (ianmcorvidae) [reply src/components/search/dataSearchQueryBuilder.js:23]: Is this right? I thought that for logged-out users we still wanted to limit to community data only.

**Review by ianmcorvidae — CHANGES_REQUESTED** [review]
I think the main bug isn't fixed (since logged-out users can still see things shared with anonymous on the detailed search page, even though they now won't be able to in the global search field), but everything else looks like it's correct/an improvement.

**Review by ianmcorvidae — APPROVED** [review]
Looks good to me!

File: src/components/search/GlobalSearchField.js [hunk 2]
```diff
@@ -365,10 +370,33 @@ function GlobalSearchField(props) {
...
+                homeDir = homeDir + "/";
+                setUserHomeDir(homeDir);
+            }
+        },
+        [bootstrapCache]
+    );
+
+    const { isFetching: isBootStrapFetching } = useBootStrap(
+        bootstrapQueryEnabled,
+        (respData) => preProcessData(respData),
```
Comment (psarando) [inline src/components/search/GlobalSearchField.js:387]:
```suggestion
        preProcessData,
```

**Review by psarando — APPROVED** [review]
LGTM 👍

**Comment by sri2k1us** [comment]
Thanks for the reviews.

**Pull request merged**